geth/mailservice/mailservice_api.go

```diff
@@ -16,52 +18,130 @@ import (
 const (
 	// defaultWorkTime is a work time reported in messages sent to MailServer nodes.
 	defaultWorkTime = 5
+
+	// gcTimeout defines timeout to release not used connection with mailserver peers.
+	gcTimeout = 5 * time.Minute
+	// gcPeriod defines how often to run garbage collector.
+	gcPeriod = 3 * time.Second
+
+	// peerConnectTimeout defines timeout to connect with a peer.
+	peerConnectTimeout = 10 * time.Second
 )
 
 var (
 	// ErrInvalidMailServerPeer is returned when it fails to parse enode from params.
 	ErrInvalidMailServerPeer = errors.New("invalid mailServerPeer value")
 	// ErrInvalidSymKeyID is returned when it fails to get a symmetric key.
 	ErrInvalidSymKeyID = errors.New("invalid symKeyID value")
+	// ErrNoServersInConfig is returned when no trusted peers are configured for p2p.Server
+	ErrNoServersInConfig = errors.New("no mailservers are available")
 )
 
 // PublicAPI defines a MailServer public API.
 type PublicAPI struct {
-	provider ServiceProvider
+	service *MailService
+
+	newConnectedPeers chan *discover.Node
```

Comment:
Maybe we can call it `selectedPeers` as it is usually used after `api.choosePeer()`?